README.md
<!-- 
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages). 

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages). 
-->

It helps developers to show animated faded image list in the application

## Features

you have to pass list of images and it has property of fit for scale

## Getting started

simply call ImageSlider() and pass list of Strings url

### Show some :heart: and star the repo to support the project


## Screenshots
<img src="https://raw.githubusercontent.com/Rajkumar07793/animated_image_list/master/image.gif"/>


## Usage

```dart

import 'package:animated_image_list/animated_image_list.dart';
import 'package:faker/faker.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

List<String> images = [
  faker.image.image(),
];

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    var faker = Faker();
    String image = faker.image.image();
    images.add(
        'https://images.unsplash.com/photo-1659993890273-e56bdc2b720f?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=480&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY2MjYzMTU4Mw&ixlib=rb-1.2.1&q=80&w=640');
    images.add(
        'https://images.unsplash.com/photo-1660335767235-9a4988cc7451?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=480&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY2MjYzMTU2MQ&ixlib=rb-1.2.1&q=80&w=640');
    images.add(image);
    // print(image);
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Animated image list',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: ImageSlider(
        images: images,
      ),
    );
  }
}

```

## Additional information

I am welcoming all the contributors to add functionalities in this package

mail me if any query

Rajkumar07793@gmail.com

CHANGELOG.md
## 0.0.1

[0.0.1] - 08/09/2022_04:00 PM.

initial release.

test/animated_image_list_test.dart
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('adds one to input values', () {
    // final calculator = ImageSlider();
    // expect(calculator.addOne(2), 3);
    // expect(calculator.addOne(-7), -6);
    // expect(calculator.addOne(0), 1);
  });
}


lib/animated_image_list.dart
library animated_image_list;

import 'package:flutter/material.dart';

class ImageSlider extends StatefulWidget {
  final List<String> images;
  final BoxFit? fit;
  const ImageSlider({Key? key, required this.images, this.fit = BoxFit.none})
      : super(key: key);

  @override
  State<ImageSlider> createState() => _ImageSliderState();
}

class _ImageSliderState extends State<ImageSlider>
    with SingleTickerProviderStateMixin {
  Animation<double>? animation;
  AnimationController? animationController;
  List? images = [];
  int imageIndex = 0;

  imgIndex() {
    if (imageIndex < images!.length - 1) {
      imageIndex++;
    } else {
      imageIndex = 0;
    }
    // print("$imageIndex }");
  }

  animate() {
    animationController =
        AnimationController(vsync: this, duration: const Duration(seconds: 4));
    CurvedAnimation curve =
        CurvedAnimation(parent: animationController!, curve: Curves.linear);
    animation = Tween(begin: 0.0, end: 1.0).animate(curve)
      ..addListener(() {
        setState(() {});
      })
      ..addStatusListener((status) {
        switch (status) {
          case AnimationStatus.completed:
            animationController!.reverse();
            Future.delayed(const Duration(milliseconds: 200)).whenComplete(() {
              imgIndex();
              imageIndex = imageIndex;
            });
            break;
          case AnimationStatus.dismissed:
            animationController!.forward();
            break;
          case AnimationStatus.forward:
            break;
          case AnimationStatus.reverse:
            break;
        }
      });
  }

  setAds() {
    setState(() {
      images = widget.images;
      animationController!.forward();
      imgIndex();
      imageIndex = imageIndex;
      // print("imageIndex $imageIndex");
    });
  }

  @override
  void initState() {
    animate();

    setAds();
    super.initState();
  }

  @override
  void dispose() {
    animationController!.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    double shortestSide = MediaQuery.of(context).size.shortestSide;
    double height = MediaQuery.of(context).size.height;
    return (images!.isNotEmpty)
        ? AnimatedOpacity(
            curve: Curves.fastOutSlowIn,
            duration: const Duration(seconds: 1),
            opacity: (animationController!.value <= 0.9) ? 1 : 0,
            child: ClipRRect(
              borderRadius: BorderRadius.circular(0),
              child: Image.network(
                images![imageIndex],
                fit: widget.fit,
                loadingBuilder: loadingBuilder,
              ),
            ),
          )
        : SizedBox(
            height: (shortestSide < 550) ? shortestSide * 1 : height * 0.6,
            child: const Center(
              child: CircularProgressIndicator(),
            ),
          );
  }

  Widget loadingBuilder(
      BuildContext context, Widget child, ImageChunkEvent? loadingProgress) {
    if (loadingProgress == null) {
      return child;
    }
    return Center(
      child: CircularProgressIndicator(
        value: loadingProgress.expectedTotalBytes != null
            ? loadingProgress.cumulativeBytesLoaded /
                loadingProgress.expectedTotalBytes!
            : null,
      ),
    );
  }
}


example/lib/main.dart
import 'package:animated_image_list/animated_image_list.dart';
import 'package:faker/faker.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

List<String> images = [
  faker.image.image(),
];

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    var faker = Faker();
    String image = faker.image.image();
    images.add(
        'https://images.unsplash.com/photo-1659993890273-e56bdc2b720f?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=480&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY2MjYzMTU4Mw&ixlib=rb-1.2.1&q=80&w=640');
    images.add(
        'https://images.unsplash.com/photo-1660335767235-9a4988cc7451?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=480&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY2MjYzMTU2MQ&ixlib=rb-1.2.1&q=80&w=640');
    images.add(image);
    // print(image);
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Animated image list',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: ImageSlider(
        images: images,
      ),
    );
  }
}
